### PR TITLE
MINOR: Replacing for with foreach loop in stream test classes

### PR DESCRIPTION
**Pull request by kashyap-prabhat**

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks for the PR, this is a good readability improvement. Surprising that there are so many old style loops in the code. @dguy, maybe you can review?

File: streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoinTest.java [hunk 1]
```diff
@@ -120,8 +120,8 @@ public void testJoin() throws Exception {
         // --> w1 = { 0:X0, 1:X1, 0:X0, 1:X1, 2:X2, 3:X3 }
         //     w2 = { 0:Y0, 1:Y1 }
 
-        for (int i = 0; i < expectedKeys.length; i++) {
-            driver.process(topic1, expectedKeys[i], "X" + expectedKeys[i]);
+        for (int expectedKey1 : expectedKeys) {
```
Comment (dguy) [inline streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoinTest.java]: nit: `expectedKey`

**Review by dguy — COMMENTED** [review]
Thanks @PKOfficial - one nit pick,  but otherwise LGTM

**Comment by kashyap-prabhat** [comment]
@dguy Thanks for pointing it out. It was a mistake. Please check I have fixed it.

**Review by dguy — APPROVED** [review]
LGTM

**Comment by kashyap-prabhat** [comment]
@IJuma Do you suggest any further improvements?

**Comment by guozhangwang** [comment]
test this please

**Comment by guozhangwang** [comment]
Thanks for the patch @PKOfficial ! LGTM and merged to trunk.

**Pull request closed without merging**